### ISSUE TITLE
Fix preprocessor info on auxiliary variables

### ISSUE THREAD

**Issue by JohannesPfeifer** [DynareTeam/dynare#773]
See #770

**Comment by houtanb** [comment]
See notes in ```preprocessor/SymbolTable.hh```
```eq_nbr``` is only set for ramsey multipliers
```orig_lead_lag``` is only set for lagged variables, despite the name. I don't know why as I didn't write this code, but the information is only tracked for lagged variables
```orig_index``` again, it seems to have been a choice not to print this.

@JohannesPfeifer, are these needed or did you just make this ticket because the information seemed incomplete?

**Comment by JohannesPfeifer** [comment]
@houtanb Yes, you are right with ```eq_nbr```. Forget about this.
It would be very helpful to generally set ```orig_lead_lag``` and ```orig_index``` as printing of error messages relies on this information. Currently, it is for example almost impossible to identify to which variable a replaced expectation operator refers to (type 4)

**Comment by houtanb** [comment]
@JohannesPfeifer for an expectation operator, it's not clear to me what ```orig_lead_lag``` and ```orig_index``` should contain when the ```expectation``` is of an expression. Can you tell me what the  output should be when you have ```expectation(-3)(x(+2)*y(+1)-y(-1))```

**Comment by JohannesPfeifer** [comment]
@houtanb I completely forgot that you can apply this to expression. This makes my initial suggestion unfeasible. My proposal would be to add a new field to ```M_.aux_vars``` that contains the replaced expression, in your example ```expectation(-3)(x(+2)*y(+1)-y(-1))``` (if that is not too much work).

**Comment by houtanb** [comment]
This is certainly doable, but it is a bit of work.... I'll leave the ticket open and get to it some day (maybe soon)

**Comment by JohannesPfeifer** [comment]
@houtanb I would not consider this as closed. You correctly pointed out that auxiliary variables can be nested. An example is ```test1.mod``` in ```tests/auxiliary_variables```. The first aux_var is the complex expression. After that, we have several nested aux_vars for the leads of this expression. Would it be possible to assign the respective ```endo_index``` from ```M_.aux_vars``` to the ```orig_index``` field of ```M_.aux_vars```? That is, the lag/lead of an auxiliary variable gets the index of the auxiliary variable it belongs to as ```orig_index```. This would allow to track the relation between variables in ```M_.endo_names```

**Comment by MichelJuillard** [comment]
When there are leads on more than one period, the auxiliary variable replaces the entire non-linear expression. For leads, there is no one to one correspondence between auxiliary variable and lead original variable

**Comment by JohannesPfeifer** [comment]
@MichelJuillard I see, but how can we then deal with this when displaying decision rules?

**Comment by MichelJuillard** [comment]
@JohannesPfeifer the current way is to use the name of the auxiliary variable

**Comment by JohannesPfeifer** [comment]
No, sometimes, we produce an error, see #1367. Would it be possible to do the same treatment as for leads as in 025e7a13c320822463c9ca19ba76112daf4d1d20 for expectations?
